code/parachain/frame/assets-transactor-router/src/tests/routes.rs

```diff
@@ -0,0 +1,1978 @@
+					.expect("string is within bound"),
+			),
+			symbol: None,
+			decimals: Some(12),
+			ratio: None,
+			existential_deposit: 100,
+		};
+
+		new_test_ext().execute_with(|| {
+			let asset_id_local = Pallet::<Test>::create_local_asset(
+				protocol_id_local,
+				nonce_local,
+				asset_info_local,
+			)
+			.unwrap();
+
+			let foreign_asset_id = XcmAssetLocation(xcm::v2::MultiLocation::parent());
+			let foreign_asset_info = AssetInfo {
+				name: Some(
+					BiBoundedAssetName::from_vec(b"Kusama".to_vec())
+						.expect("string is within bound"),
+				),
+				symbol: Some(
+					BiBoundedAssetSymbol::from_vec(b"KSM".to_vec())
+						.expect("string is withing bound"),
+				),
+				decimals: Some(12),
+				ratio: None,
+				existential_deposit: 1000,
+			};
+
+			let asset_id_foreign = Pallet::<Test>::create_foreign_asset(
+				protocol_id_foreign,
+				nonce_foreign,
+				foreign_asset_info,
+				foreign_asset_id,
+			)
+			.unwrap();
+
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::minimum_balance(NATIVE_ASSET_ID),
+				1
+			);
+			// should be 0 because of ExistentialDeposits for orml_tokens
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::minimum_balance(asset_id_local),
+				0
+			);
+			// should be 0 because of ExistentialDeposits for orml_tokens
+			assert_eq!(
+				<Pallet::<Test> as MultiCurrency<AccountId>>::minimum_balance(asset_id_foreign),
+				0
+			);
+
+			assert_eq!(<<Test as Config>::NativeTransactor>::minimum_balance(), 1);
+			assert_eq!(<<Test as Config>::LocalTransactor>::minimum_balance(asset_id_local), 0);
+			assert_eq!(<<Test as Config>::ForeignTransactor>::minimum_balance(asset_id_foreign), 0);
+		});
+	}
+
+	#[test]
+	fn total_issuance() {
+		let protocol_id_local = *b"testloca";
+		let nonce_local = 0;
+		let protocol_id_foreign = *b"testfore";
+		let nonce_foreign = 0;
+		let asset_info_local = AssetInfo {
+			name: Some(
+				BiBoundedAssetName::from_vec(b"local asset".to_vec())
+					.expect("string is within bound"),
+			),
+			symbol: None,
+			decimals: Some(12),
+			ratio: None,
+			existential_deposit: 100,
+		};
+
+		new_test_ext().execute_with(|| {
+			let asset_id_local = Pallet::<Test>::create_local_asset(
+				protocol_id_local,
+				nonce_local,
+				asset_info_local,
+			)
+			.unwrap();
+
+			let foreign_asset_id = XcmAssetLocation(xcm::v2::MultiLocation::parent());
+			let foreign_asset_info = AssetInfo {
+				name: Some(
+					BiBoundedAssetName::from_vec(b"Kusama".to_vec())
+						.expect("string is within bound"),
+				),
+				symbol: Some(
+					BiBoundedAssetSymbol::from_vec(b"KSM".to_vec())
+						.expect("string is withing bound"),
+				),
+				decimals: Some(12),
+				ratio: None,
+				existential_deposit: 1000,
+			};
+
+			let asset_id_foreign = Pallet::<Test>::create_foreign_asset(
+				protocol_id_foreign,
+				nonce_foreign,
+				foreign_asset_info,
+				foreign_asset_id,
+			)
+			.unwrap();
```

Comment:
you seem to be using the same asset info in several tests, could this be extracted into a helper function?